packages/react-query/src/__tests__/prefetch.test.tsx

```diff
@@ -0,0 +1,352 @@
+
+import type { InfiniteData, UseInfiniteQueryOptions, UseQueryOptions } from '..'
+import type { Mock } from 'vitest'
+
+const generateQueryFn = (data: string) =>
+  vi.fn<any, Promise<string>>().mockImplementation(async () => {
+    await sleep(10)
+
+    return data
+  })
+
+const generateInfiniteQueryOptions = (
+  data: Array<{ data: string; currentPage: number; totalPages: number }>,
+) => {
+  let currentPage = 0
+
+  return {
+    queryFn: vi
+      .fn<any, Promise<(typeof data)[number]>>()
+      .mockImplementation(async () => {
+        const currentPageData = data[currentPage]
+        if (!currentPageData) {
+          throw new Error('No data defined for page ' + currentPage)
+        }
+
+        await sleep(10)
+        currentPage++
+
+        return currentPageData
+      }),
+    initialPageParam: 1,
+    getNextPageParam: (lastPage: (typeof data)[number]) =>
+      lastPage.currentPage === lastPage.totalPages
+        ? undefined
+        : lastPage.currentPage + 1,
+  }
+}
+
+describe('usePrefetchQuery', () => {
+  const queryCache = new QueryCache()
+  const queryClient = createQueryClient({ queryCache })
+
+  function Suspended<TData = unknown>(props: {
+    queryOpts: UseQueryOptions<TData, Error, TData, Array<string>>
+  }) {
+    const state = useSuspenseQuery(props.queryOpts)
+
+    return (
+      <div>
+        <div>data: {String(state.data)}</div>
+      </div>
+    )
+  }
+
+  it('should prefetch query if query state does not exist', async () => {
+    const queryOpts = {
+      queryKey: queryKey(),
+      queryFn: generateQueryFn('Prefetch is nice!'),
+    }
+
+    function App() {
+      usePrefetchQuery(queryOpts)
+
+      return (
+        <React.Suspense fallback="Loading...">
+          <Suspended queryOpts={queryOpts} />
+        </React.Suspense>
+      )
+    }
+
+    const rendered = renderWithClient(queryClient, <App />)
+
+    await waitFor(() => rendered.getByText('data: Prefetch is nice!'))
+    expect(queryOpts.queryFn).toHaveBeenCalledTimes(1)
+  })
+
+  it('should not prefetch query if query state exists', async () => {
+    const queryOpts = {
+      queryKey: queryKey(),
+      queryFn: generateQueryFn('The usePrefetchQuery hook is smart!'),
+    }
+
+    function App() {
+      usePrefetchQuery(queryOpts)
+
+      return (
+        <React.Suspense fallback="Loading...">
+          <Suspended queryOpts={queryOpts} />
+        </React.Suspense>
+      )
+    }
+
+    await queryClient.fetchQuery(queryOpts)
+    ;(queryOpts.queryFn as Mock).mockClear()
+    const rendered = renderWithClient(queryClient, <App />)
+
+    expect(rendered.queryByText('fetching: true')).not.toBeInTheDocument()
+    await waitFor(() =>
+      rendered.getByText('data: The usePrefetchQuery hook is smart!'),
+    )
+    expect(queryOpts.queryFn).not.toHaveBeenCalled()
+  })
+
+  it('should display an error boundary if query cache is populated with an error', async () => {
```

Comment:
NIT: I find this slightly misleading since it's the `useSuspenseQuery` that triggers the error that shows the error boundary.

Not perfect, but maybe something along the lines of:

```suggestion
  it('should let errors fall through and not refetch failed queries', async () => {
```